tsdb/exemplar.go

```diff
@@ -0,0 +1,33 @@
+package tsdb
+
+import (
+	"github.com/prometheus/prometheus/pkg/exemplar"
+	"github.com/prometheus/prometheus/pkg/labels"
+)
+
+type ExemplarStorage struct {
+	exemplars map[uint64][]exemplar.Exemplar
```

Comment:
I'd use a `map[string][]exemplar.Exemplar` and use `Labels.String()` rather than `Labels.Hash()` to create the key. It has no collisions and should only be marginally slower. (Hash maps of strings are highly optimized.) Then you can also address Brian's comment about not having a hash in the API, and not having a collision problem.

Reply:
This would be great to benchmak. last time I tried, it was a lot slower.